app/modules/auth/auth.factory.js

```diff
@@ -22,18 +22,22 @@ function authFactory($location, $rootScope, $state, $transitions, AnnotationsFac
             zooAPI.beforeEveryRequest = function() {
                 return OAuth.checkBearerToken()
                     .then(function (token) {
-                        // The Panoptes client doesn't return an error but just null when it can't refresh the token.
-                        // So we check for null, instead of using a catch block.
-
-                        if (_user.id && token === null) {
-                            // We are logged in but don't have a token any more.
-                            alert('Your session is expired. Press OK to save your work and start a new one.')
-                            // Save any unsaved work and redirect to Panoptes for a new token.
-                            // AnnotationsFactory.updateCache();
-                            // OAuth.signIn($location.absUrl());
-                            return Promise.reject(new Error('HELP!'));
-
-                        }
+                        // We catch any request to access the transcribe route.
+                        $transitions.onStart({ to: 'Transcribe'}, function(transition) {
```

Comment:
What happens here if my session expires while I'm navigating to a different page, such as the home page, or if I'm already transcribing when my session expires (which is the most likely scenario)? Correct me if I'm wrong, but this block looks like it only checks if I'm still logged in when I'm about to load the transcription page.

Reply:
You're right. Removing the `Transcribe` restriction will take care of navigating e.g. from `/transcribe` to home `/`. However, if my session expires while I'm transcribing and I click `I'm done`, it'll just look like I'm still logged in and new subject will be served, which isn't right. I'll have to catch the event too.

Reply:
You want to check for a valid token before every API request anyway, so just remove this check.